### ISSUE TITLE
filter all and any feature

### ISSUE THREAD

**Issue by Deninc** [sparklyr/sparklyr#518]
Hi,

I wonder if there is any equivalent of **filter all** and **any** feature in sparklyr. For example:
```
  id value
1  1    32
2  1    NA
3  1    45
4  2    NA
5  2    NA
6  3    19
7  3   122
8  3   100
```
`df %>% group_by(id) %>% filter(all(is.na(value)))`
```
     id value
  <dbl> <dbl>
1     2    NA
2     2    NA
```

**Comment by kevinushey** [comment]
Here's a somewhat roundabout way of expressing something like that:

```R
library(sparklyr)
library(dplyr)

df <- data.frame(
  id = c(1, 1, 1, 2, 2, 3, 3, 3),
  value = c(32, NA, 45, NA, NA, 19, 122, 100)
)

sc <- spark_connect(version = "2.1.0", master = "local")
tbl <- copy_to(sc, df, overwrite = TRUE)

tbl %>%
  group_by(id) %>%
  filter(
    sum(as.numeric(is.na(value))) == n()
  )
```

Giving:

```
Source:   query [2 x 2]
Database: spark connection master=local[8] app=sparklyr local=TRUE
Groups: id

     id value
  <dbl> <dbl>
1     2   NaN
2     2   NaN
```

**Comment by Deninc** [comment]
@kevinushey thanks for your suggestion. 

It would be nice to have such feature, **all** and **any** are very intuitive and they don't necessarily need to loop through all values.

**Comment by yitao-li** [comment]
Update: actually ignore https://github.com/sparklyr/sparklyr/pull/2925 as it doesn't implement the `all()` / `any()` semantic you'd want following a `dplyr::group_by()`.

The closest you can get is https://spark.apache.org/docs/latest/api/sql/index.html#bool_and and https://spark.apache.org/docs/latest/api/sql/index.html#bool_or (only supported in Spark 3.0), but still, those 2 end up collapsing all records within a group instead of allowing you to inspect individual records per group.

So, TL;DR: AFAIK there is still no simple way to translate this type of query into Spark SQL
